### ISSUE TITLE
Loading ... on refresh.  SSR is not working right.

### ISSUE THREAD

**Issue by justinbmeyer** [donejs/bitballs#137]
This happens in development mode too.  

To reproduce:

- Login (it doesn't matter, but is easier to spot)
- Go to http://localhost:5000/players
- Refresh the page.  

You'll see the server-side rendered content, a flash of "loading", and then the content again.

My guess is that can-wait isn't properly waiting for the progressive loading parts to finish.

cc @matthewp @jandjorgensen @James0x57

**Comment by James0x57** [comment]
If I can help with anything - setup or whatever, let me know!

**Comment by matthewp** [comment]
@justinbmeyer What do you mean by step 3 "Refresh Login", what do I need to do here to recreate the problem?

**Comment by matthewp** [comment]
I think this is caused because of can-connect importing its own Promise implementation. So we have to Promises floating around are only overriding the global one when we need to be overriding both.

**Comment by matthewp** [comment]
I started https://github.com/canjs/can-zone/issues/57 to discuss the global idea we discussed. It might be worth doing to handle the local reference case.  I would estimate 2 days of work.

However it would not fix this particular issue because Steal is using its own Promise which is within its closure, so if we override Promise globally, after Steal is loaded, our code will still not be used.

Going ahead with the fix to wrap System.import.
